internal/worker/sshserver/server.go

```diff
@@ -228,18 +238,31 @@ func (s *ServerWorker) directTCPIPHandler(srv *ssh.Server, conn *gossh.ServerCon
 		},
 	}
 
-	// TODO(ale8k): Update later to generate host keys per unit.
-	terminatingHostKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	signer, err := s.getHostKeySignerForTarget(d.DestAddr)
```

Comment:
No need for `get` in getter.
```suggestion
	signer, err := s.hostKeySignerForTarget(d.DestAddr)
```